action.yml

```diff
@@ -0,0 +1,37 @@
+name: Add pull request to project
+description: GitHub Action to add new pull requests to the Community Contributions GitHub project
+
+runs:
+  using: composite
+  steps:
+
+    - name: Check if should be added
+      id: check-if-should-add
+      env:
+        PR_AUTHOR: ${{ github.event.pull_request.user.login }}
+      run: |
+        # We don't want CMS Squad member PRs to clutter the project board
+        echo "Author is $PR_AUTHOR"
+        for squad_member in 'GuySartorelli' 'emteknetnz' 'maxime-rainville'; do
+          if [[ $PR_AUTHOR == $squad_member ]]; then
+            echo "Author is in CMS Squad. Skipping."
+            echo "should_add_to_project=false" >> $GITHUB_OUTPUT
+            exit 0
+          fi
+        done
+        echo "should_add_to_project=true" >> $GITHUB_OUTPUT
+
+    - name: Generate token
+      id: generate-token
+      if: steps.check-if-should-add.outputs.should_add_to_project == 'true' && github.repository_owner == 'silverstripe'
+      uses: actions/create-github-app-token@v1
+      with:
+        app-id: ${{ vars.PROJECT_MANAGER_APP_ID }}
+        private-key: ${{ secrets.PROJECT_MANAGER_PRIVATE_KEY }}
+  
+    - name: Add to project
+      uses: actions/add-to-project@v1.0.1
+      with:
+        # Add to the Community Contributions project
+        project-url: https://github.com/orgs/silverstripe/projects/4
+        github-token: ${{ steps.generate-token.outputs.token }}
```

Comment:
Have you double checked this shows absolutely nowhere in the log output?

Reply:
This is the officially documented way to do this - I would be very shocked if they printed the token to the log in the official action.

After this is merged we can test it out and check log output - I don't think it's worth doing before hand because I very much doubt GitHub would print tokens to the log.